src/main/java/com/taskadapter/redmineapi/AttachmentManager.java

```diff
@@ -7,198 +14,144 @@
-                                       InputStream content) throws RedmineException, IOException {
-        final InputStream wrapper = new MarkedInputStream(content,
-                "uploadStream");
-        final String token;
-        try {
-            token = transport.upload(wrapper);
-            final Attachment result = AttachmentFactory.create();
-            result.setToken(token);
-            result.setContentType(contentType);
-            result.setFileName(fileName);
-            return result;
-        } catch (RedmineException e) {
-            unwrapException(e, "uploadStream");
-            throw e;
-        }
-    }
+	/**
+	 * Delivers an {@link com.taskadapter.redmineapi.bean.Attachment} by its ID.
+	 * 
+	 * @param attachmentID
+	 *            the ID
+	 * @return the {@link com.taskadapter.redmineapi.bean.Attachment}
+	 * @throws RedmineAuthenticationException
+	 *             thrown in case something went wrong while trying to login
+	 * @throws RedmineException
+	 *             thrown in case something went wrong in Redmine
+	 * @throws NotFoundException
+	 *             thrown in case an object can not be found
+	 */
+	public Attachment getAttachmentById(int attachmentID) throws RedmineException {
+		return transport.getObject(Attachment.class, attachmentID);
+	}
 
-    /**
-     * Delivers an {@link com.taskadapter.redmineapi.bean.Attachment} by its ID.
-     *
-     * @param attachmentID the ID
-     * @return the {@link com.taskadapter.redmineapi.bean.Attachment}
-     * @throws RedmineAuthenticationException thrown in case something went wrong while trying to login
-     * @throws RedmineException        thrown in case something went wrong in Redmine
-     * @throws NotFoundException       thrown in case an object can not be found
-     */
-    public Attachment getAttachmentById(int attachmentID) throws RedmineException {
-        return transport.getObject(Attachment.class, attachmentID);
-    }
+	public void downloadAttachmentContent(Attachment issueAttachment, OutputStream stream) throws RedmineException {
+		transport.download(issueAttachment.getContentURL(), new CopyBytesHandler(stream));
+	}
 
-    public void downloadAttachmentContent(Attachment issueAttachment,
-                                          OutputStream stream) throws RedmineException {
-        transport.download(issueAttachment.getContentURL(),
-                new CopyBytesHandler(stream));
-    }
+	/**
+	 * Downloads the content of an {@link com.taskadapter.redmineapi.bean.Attachment} from the Redmine server.
+	 * 
+	 * @param issueAttachment
+	 *            the {@link com.taskadapter.redmineapi.bean.Attachment}
+	 * @return the content of the attachment as a byte[] array
+	 * @throws RedmineCommunicationException
+	 *             thrown in case the download fails
+	 */
+	public InputStream downloadAttachmentContent(Attachment issueAttachment) throws RedmineException {
+		final ByteArrayOutputStream baos = new ByteArrayOutputStream();
```

Comment:
And this creates a new error where no big files could be downloaded from the redmine. Why do you want to store an attachement in the memory? Why could you store an attachement in the memory? It could be a quite large file.

This method could cause OutOfMemory errors. And it will be good to have these Errors in this method and not in other (unrelated) parts of the program.

Original input stream approach do not have this problem. Library user is able to deal with it. If he/she want to copy an attachement into the file output stream, he/she is able to do so with the original version.